halo2/src/prover.rs

```diff
@@ -132,27 +130,16 @@ impl Halo2Prover {
         Ok(proof)
     }
 
-    pub fn prove_aggr<F: FieldElement>(
+    pub fn prove_aggr(
         &self,
-        pil: &Analyzed<F>,
-        fixed: &[(String, Vec<F>)],
         witness: &[(String, Vec<F>)],
         proof: Vec<u8>,
     ) -> Result<Vec<u8>, String> {
         log::info!("Starting proof aggregation...");
 
-        // TODO this is hacky
-        let degree = usize::BITS - pil.degree().leading_zeros() + 1;
-        let params_app = {
-            let mut params = self.params.clone();
-            params.downsize(degree);
-            params
-        };
```

Comment:
Why was this removed? This is important because the universal setup may be a common large one, and here we want to downsize it to the circuit's degree

Reply:
I have placed this `downsize()` in `create()`. Do you think it is correct there? Still fails the test.

Reply:
which test?

Reply:
The one that was failing. But it is fixed now.